docs/content/sips/xxx-admin-api.md

```diff
@@ -0,0 +1,92 @@
+title = "SIP xxx - The Admin API
+template = "main"
+date = "2022-06-07T22:49:37"
+
+---
+
+Summary: An Administration API for Spin.
+
+Owner: me@mitchellhynes.com
+
+Created: June 7th, 2022
+
+Updated: June 7th, 2022
+
+## Background
+
+Apart from reading the file system of the host machine, there's no way to read
+logs or manifest of a running Spin app. This means users are limited in what
+they know about a Spin app while it's running, unless they have access to the
+environment it is on.
+
+[Laundromat](https://github.com/ecumene/laundromat) is a tool for reading logs
+and organizing them in a quicker format for developer velocity, and it gets
+around this by serving the logs of a Spin app as a REST API. This is limited
+because
+
+- it must have access to the file system which users may not have access to
+- being a server itself it must be forwarded out of whatever network users are
+  running it in, along with Spin itself
+- the file system is itself limited as new logs won't notify the user.
+  system file events could be used but they're notoriously platform dependent.
+
+If Spin exposed an "administrator's API" that is handled by the same route that
+Spin apps are served on we could open the door to many hypervisor-like tools.
+
+## Proposal
```

Comment:
Do you see anything that would prevent us from exposing any of this data as a host-provided Wasm interface (i.e. host component)? All of this API would still be relevant, but could be implemented as http trigger(s). One advantage of this approach would be that other trigger types - notably the upcoming "clock trigger" - could access the same info.

Reply:
No, in fact I think that might be better for my use case entirely.

I'm going to rewrite the Spin to account for this :)